### ISSUE TITLE
Notifications (end user) related to the app workflow/events

### ISSUE THREAD

**Issue by lvbachmann** [Altinn/app-lib-dotnet#46]
## Introduction
It should be possible to connect notifications by email or sms to certain events or workflow steps happening in an app. An example: When an app moves from a feedback-state to an archived-state, the system could send an email notifying the end user that the status has changed.

We should probably allow app owner to define some kinds of logic to add notifications, i.e. form has been instantiated but hasn't been used for 14 days.

## In scope
> What's in scope of this Epic?

TBD

## Out of scope
> What's **out** of scope for this Epic?

TBD

## Product requirement
> Give an understanding of what should be designed, built, tested and released.

TBD

## Design requirement
> UX and other requirements for the end result.

TBD

## Technical/Engineering requirement
> How to build the feature technically, APIs to use/build, database storage etc?

TBD

## Related issues
- Blocked by: https://github.com/Altinn/altinn-notifications/issues/148

**Comment by JanAage99** [comment]
OED har behov for at varsel inneholder lenke til OED-app slik at bruker/arving kan starte app på smidig måte. 
OED sin WI:
https://github.com/Altinn/oed/issues/142

**Comment by FinnurO** [comment]
Feature complete in the sense you can notify a user either when delegating a task or a task you don't have the authorisation to perform.

**Comment by RonnyB71** [comment]
Blocked by https://github.com/Altinn/altinn-notifications/issues/1

**Comment by RonnyB71** [comment]
Closing this one as it is not specific in the case it needs to solve, and we don't have specific requirements to be able to configure type of notification described in the headline. When Altinn/dialogporten-frontend#4402 is implemented we will have a notification client allowing for notifications to be sent from anywhere within the app, including the various app events. And hence it can be solved by using this client and plug the notification code where you would like.
